lib/Base/Image.cpp

```diff
@@ -270,6 +286,48 @@ bool glow::writePngImage(Tensor *T, const char *filename,
   return false;
 }
 
+Tensor glow::readPngImageAndPreprocess(const std::string &filename,
+                                       ImageNormalizationMode imageNormMode,
+                                       ImageChannelOrder imageChannelOrder,
+                                       ImageLayout imageLayout,
+                                       bool useImagenetNormalization) {
+  // Get image dimensions and check if grayscale or color.
+  size_t imgHeight;
+  size_t imgWidth;
+  bool isGray;
+  std::tie(imgHeight, imgWidth, isGray) = getPngInfo(filename.c_str());
```

Comment:
I don't think it's necessary to call `getPngInfo()` here -- you can get `imgHeight`, `imgWidth`, and `numChannels` from the shape of `imageData` below.